### PR TITLE
feat!: updated the default context manager to be AsyncLocalContextManager

### LINKED ISSUE #1724: Make async local context manager default
### Description
We introduced an async context manager that leverages AsyncLocalStorage in 9.4.0.  We no longer want to maintain the legacy context manager that instrumented the event loop.

### Acceptance Criteria
_What tasks need to be accomplished to achieve the goal?_
As a Node.js new relic user I want the ability to see traces for my async apps so that I can gain insight into its performance.

### Additional context
Make async local feature flag default to true.  and keep legacy context manager. Make sure we call out in release notes that the legacy is going away in the next major release.

### PR DESCRIPTION
**Pull request by mrickard**
## Description

* Made AsyncLocalContextManager the default context manager
* Gated the old context manager behind the legacy_context_manager feature flag

## How to Test

Context managers are used in both unit and integration tests. Previously, async local context was tested with scripts enabling its feature flag; this has been removed, and the legacy context manager now is testable via `unit:legacy-context`, `integration:legacy-context`, `versioned:legacy-context`, and `versioned:legacy-context:major` scripts.

Currently a draft PR: I may need to rebase to pick up CI and other changes from Node 20 branch

## Related Issues

Closes #1724 
Closes NR-141994

### CODE REVIEW

**Comment by mrickard** [comment]
Defaulting to the async context manager caused many failures in Node 14 versioned tests. The context manager selector has a conditional to coerce Node 14 to use the legacy context manager. In some unit and integration tests, that appears to be bypassed, so those tests have a similar conditional to specify the legacy manager.

File: test/integration/core/zlib.tap.js [hunk 1]
```diff
@@ -10,6 +10,11 @@ const zlib = require('zlib')
 const helper = require('../../lib/agent_helper')
 const verifySegments = require('./verify')
 const concat = require('concat-stream')
+const semver = require('semver')
+
+const legacyCtxOnly =
```
Comment (bizob2828) [inline test/integration/core/zlib.tap.js:15]: I'm unclear why this is here, can you expand?

File: documentation/feature-flags.md [hunk 2]
```diff
@@ -39,8 +39,8 @@ Any prerelease flags can be enabled or disabled in your agent config by adding a
...
 
-#### async_local_context
+#### legacy_context_manager
 * Enabled by default: `false`
-* Configuration: `{ feature_flag: { async_local_context: true|false }}`
-* Environment Variable: `NEW_RELIC_FEATURE_FLAG_ASYNC_LOCAL_CONTEXT`
-* Description: This context manager leverages AsyncLocalStorage for async context propagation. Enabling this feature flag should reduce the agent's memory and CPU use.
+* Configuration: `{ feature_flag: { legacy_context_manager: true|false }}`
+* Environment Variable: `NEW_RELIC_FEATURE_FLAG_LEGACY_CONTEXT_MANAGER`
+* Description: The legacy context manager was replaced by AsyncLocalStorage for async context propagation. If your application relies on the older behavior, you may want to enable this older context manager. Enabling this feature flag may increase the agent's use of memory and CPU.
```
[the remark this answers is not in the document]
Comment (mrickard) [reply documentation/feature-flags.md:46]: This proposed change introduces a typo, but I can pick up the gist of it.

[the remark this answers is not in the document]
Comment (mrickard) [reply documentation/feature-flags.md]: I've updated the description in feature-flags.md

File: test/integration/core/fs.tap.js [hunk 3]
```diff
@@ -13,6 +13,11 @@ const helper = require('../../lib/agent_helper')
 const verifySegments = require('./verify')
 const NAMES = require('../../../lib/metrics/names')
 
+const semver = require('semver')
+const legacyCtxOnly =
```
[the remark this answers is not in the document]
Comment (mrickard) [reply test/integration/core/fs.tap.js]: This branch was initially off main before we landed the work that drops Node 14, so these tests needed to be able to run legacy as well as async. Now that we don't have that constraint, I've refactored.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**